armory/utils/metrics.py

```diff
@@ -159,6 +159,50 @@ def _snr_spectrogram(x_i, x_adv_i):
     return signal_power / noise_power
 
 
+# Metrics specific to MARS model preprocessing in video UCF101 scenario
+
+
+def verify_mars(x, x_adv):
+    if len(x) != len(x_adv):
+        raise ValueError(f"len(x) {len(x)} != {len(x_adv)} len(x_adv)")
+    for x_i, x_adv_i in zip(x, x_adv):
+        if x_i.shape[1:] != x_adv_i.shape[1:]:
+            raise ValueError(f"Shape {x_i.shape[1:]} != {x_adv_i.shape[1:]}")
+        if x_i.shape[1:] != (16, 3, 112, 112):
```

Comment:
Per Mike Tan's [comment](https://github.com/twosixlabs/armory/issues/715#issuecomment-675743273), it looks like the [Intel](https://github.com/twosixlabs/gard-submissions/blob/a94d66815c170b078827e6d06fea0b43512cd82b/repos/IntelLabs__GARD__gard-phase1-eval1/GARD/oscar/classifiers/ucf101_mars.py#L25) submission and [JHUM's custom classifier](https://github.com/twosixlabs/gard-submissions/blob/a94d66815c170b078827e6d06fea0b43512cd82b/repos/shaoyuanlo__GARD_eval1__master/GARD_eval1/ucf101_3bn.py#L364) don't directly use the MARS preprocessing. So a less strict check may make sense.